### PR TITLE
Add genesis id into config

### LINKED ISSUE #317: Wrong chain application check
- [x] Add optional genesis block to config to prevent synchronization stops
- [x] Write tests for incorrect blocks application (valid header, invalid block sections + valid full block, invalid transactions). Ensure, that application works correct after this attempts

### PR DESCRIPTION
**Pull request by daron666**
Fixes #317

### CODE REVIEW

File: src/main/resources/application.conf [hunk 1]
```diff
@@ -34,6 +34,9 @@ ergo {
 
     # Number of state snapshot diffs to keep. Defines maximum rollback depth
     keepVersions = 200
+
+    # Optional genesis id block. Defines id of a block that should be applied firstly. Uncomment to use
```
Comment (catena2w) [inline src/main/resources/application.conf]: Description: "Defines an id of the genesis block. Other genesis blocks will be considered invalid."
Put it to `chain` section and add id of current testnet genesis block to comment

File: src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala [hunk 2]
```diff
@@ -314,6 +314,10 @@ trait HeadersProcessor extends ToDownloadProcessor with ScorexLogging with Score
         .validateEqualIds(header.parentId, Header.GenesisParentId) { detail =>
           fatal(s"Genesis block should have genesis parent id. $detail")
         }
+        .validate(config.genesisId.fold(true) { _ == Algos.encode(header.id) }) {
```
Comment (catena2w) [inline src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala]: `forall` should be better here

File: src/test/scala/org/ergoplatform/nodeView/ErgoNodeViewHolderSpecification.scala [hunk 3]
```diff
@@ -355,6 +359,102 @@ class ErgoNodeViewHolderSpecification extends ErgoNodeViewHolderTestHelpers {
     }
   }
 
+  private val t12 = TestCase("Do not apply txs with wrong header id") { fixture =>
```
Comment (catena2w) [inline src/test/scala/org/ergoplatform/nodeView/ErgoNodeViewHolderSpecification.scala]: How does this relates to current PR?

Comment (daron666) [reply src/test/scala/org/ergoplatform/nodeView/ErgoNodeViewHolderSpecification.scala]: Was in issue description. Add genesis id and some test cases.

image

**Pull request merged**